etc/kayobe/dnf.yml

```
@@ -48,7 +48,12 @@
 # This is done by default for hosts in the overcloud group via a group_vars
 # file.
 stackhpc_dnf_repos_8: "{{ dnf_custom_repos_el8 | combine(lookup('vars', 'dnf_custom_repos_' ~ ansible_facts.distribution | lower )) }}"
-stackhpc_dnf_repos_9: "{{ dnf_custom_repos_el9 | combine(dnf_custom_repos_rocky_9) }}"
+stackhpc_dnf_repos_9: >-
+  {{
+    dnf_custom_repos_el9 | combine(dnf_custom_repos_rocky_9) | combine(dnf_custom_repos_elrepo_9)
+    if dnf_install_elrepo_9 else
+    dnf_custom_repos_el9 | combine(dnf_custom_repos_rocky_9)
```

FWIW I might do it like this:
```suggestion
    dnf_custom_repos_el9 | combine(dnf_custom_repos_rocky_9) | combine(dnf_custom_repos_elrepo_9 if dnf_install_elrepo_9 | bool else {})
```

---

So much nicer.